examples/next-openai-pages/pages/chat/stream-chat-files/index.tsx

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable @next/next/no-img-element */
+import { useChat } from 'ai/react';
+
+export default function Page() {
+  const {
+    messages,
+    input,
+    files,
+    handleSubmit,
+    handleInputChange,
+    handleFileInputChange,
+    isLoading,
+  } = useChat({
+    api: '/api/stream-chat',
+  });
+
+  return (
+    <div className="flex flex-col gap-2">
+      <div className="flex flex-col p-2 gap-2">
+        {messages.map(message => (
+          <div key={message.id} className="flex flex-row gap-2">
+            <div className="w-24 text-zinc-500 flex-shrink-0">{`${message.role}: `}</div>
+            {typeof message.content === 'string' ? (
+              <div className="w-full">{message.content}</div>
+            ) : (
+              <div className="flex flex-col gap-2">
+                {message.content
```

Comment:
is `message.content` currently a string? if yes, we may want to keep that for backwards compat. We could e.g. introduce `message.parts` for now and then figure out a migration path

Reply:
switched to use `parts`